Update Developer Policies (#2974)
MicrosoftDocs/edge-developer: microsoft-edge/extensions-chromium/store-policies/developer-policies.md

Author: todo: click every link in rendered article and update the link text to match destination title|heading.

###### 1.1.8 Altering browser settings

Your extension must not, without appropriate user consent, alter, or appear to alter, browser functionality or settings including, but not limited to: the address bar search provider and suggestions, the start or home page, the new tab page, and adding or removing favorites.

Any changes made to the device settings must be done with the user's knowledge and consent, and must be easily reversible by the user.

Any alteration of browser settings, such as new tabs and search engines, should be mentioned in the listing or description, and such alterations should use standard APIs to achieve that functionality.  They must not override default settings without user permission.

<!-- ------------------------------ -->
#### 1.2 Security

Extensions must declare all permissions that are required for their functionality in the manifest.  Your extension must only request those permissions that are essential for functioning.

> Any [declarativeNetRequest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/declarativeNetRequest) rules must be clearly declared within the extension, and should not be imported remotely unless necessary for functionality.

Reviewer: remove the link from mozilla. Do not add any link here

Author: Any `declarativeNetRequest` rules must be clearly declared within

<!-- ------------------------------ -->
#### 2.12 Age ratings

###### 2.12.1 Mature content

When you submit your extension to Partner Center, you must indicate whether your extension displays content that should be marked "Mature".  When determining the rating for your extension, consider all the content in your app, including user generated content and ads, and to the content that your extension links.  If you indicate that your extension doesn't contain any "Mature" content, you are responsible for maintaining the accuracy of this rating.

Regardless of the rating given to your extension, it must still adhere to all the content requirements of the Microsoft Edge Add-ons store developer policies.

###### 2.12.2 Ratings change

If your extension provides content (such as user-generated, retail, or other web-based content) that might be appropriate for a higher age rating than the assigned rating, you must require users to opt into receiving such content by using a content filter or by signing in with a pre-existing account.

<!-- ------------------------------ -->
#### 2.13 Videos

> If you submit a promotional video in the listing, then it should strictly adhere to our content guidelines.  If you choose to provide a YouTube link, you must ensure that advertisements are turned off for the specific videos that you want to embed.  For more information on turning on or off ads on YouTube, see [YouTube advertising formats](https://support.google.com/youtube/answer/2531367?ref_topic=7072227) and [Ads on embedded videos](https://support.google.com/youtube/answer/132596).

Author: ```suggestion
If you submit a promotional video in the listing, the promotional video must strictly adhere to the Developer policies for the Microsoft Edge Add-ons store.  If you choose to provide a YouTube link, you must ensure that advertisements are turned off for the specific videos that you want to embed.  For more information on turning on or off ads on YouTube, see [YouTube advertising formats](https://support.google.com/youtube/answer/2531367?ref_topic=7072227) and [Ads on embedded videos](https://support.google.com/youtube/answer/132596).
```
"our" is bad, too personal/possessive.  reword to avoid "our"
@Bhuvana-Priya 
"it should strictly adhere to our content guidelines."
Solve by being specific.
Alts:
* it should strictly adhere to the Developer policies for the Microsoft Edge Add-ons store.
* it should strictly adhere to the content guidelines in documents x, y, and z.

Author: If you submit a promotional video in the listing, then it should strictly adhere to these content policies.

ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 12/14/2023
---

> # Microsoft Edge Add-ons store developer policies

Author: ```suggestion
Developer policies for the Microsoft Edge Add-ons store 
```
start w/ key words; avoid a long string of modifiers followed by the key words.  easier/quicker to see which article is meant

Author: @Bhuvana-Priya (@AbhisekhMSFT) - I changed the title to start with "Developer policies".

From the POV of a legal document's title, considered in isolation, one might lean towards:
Microsoft Edge Add-ons store developer policies
(stiff & formal sounding)

But from the POV of looking at article titles in the TOC, lean towards:
Developer policies for the Microsoft Edge Add-ons store
to help quickly differentiate & characterize the various articles.  Practical.

Author: The repo uses too many different phrases:
```
The _Microsoft Edge Add-ons store_<!--36--> is also called 
the _Edge Add-ons store_,<!--12--> 
the _Microsoft Edge Add-ons website_,<!--89--> or 
the _Edge Add-ons website_.<!--5-->
```
That's covered by work item 47150845 - [Extensions] Normalize name of store/website, and "add-ons" vs. "extensions"

If you make any changes to your extension beyond the described functionality, any changes to code must be compliant with the Content Security Policy for Microsoft Edge extensions; see [Content Security Policy (CSP)](./csp.md).  For example, your extension should not download a remote script and subsequently run that script in a manner that is not consistent with the described functionality.

###### 1.2.2 Unwanted and malicious software

Your extension must not contain or enable malware that may harm the operation of the network, servers, and infrastructure of Microsoft or any third parties. Spyware, malicious scripts, and phishing scams are also prohibited.

Your extension must comply with the Microsoft criteria for Unwanted and Malicious Software, listed in [How Microsoft identifies malware and potentially unwanted applications](/windows/security/threat-protection/intelligence/criteria).

###### 1.2.3 Dependency on other software

Your extension may depend on non-integrated software (such as another product, module, or service) to deliver the primary functionality. However, this information should be clearly disclosed in the description.

###### 1.2.4 Extension updates

> Unless otherwise permitted by Microsoft, your extension must be updated only through Microsoft Edge Add-ons.

Author: ```suggestion
Unless otherwise permitted by Microsoft, your extension must be updated only through the Microsoft Edge Add-ons store.
```
@Bhuvana-Priya 
Existing sentence doesn't make sense at end: add-ons must be updated through add-ons.

Author: @Bhuvana-Priya - is this what's meant?
Unless otherwise permitted by Microsoft, your extension must be updated only through Partner Center and the Edge Add-ons store.

Author: applied the change

* Content deemed not family-friendly.

* Bots: Any extension that automatically generates spam-like or unsolicited messages on the user's behalf or that which performs automated actions to manipulate a platform's features, such as artificially inflating engagement metrics.

* Cryptocurrency: Crypto mining and any other illegitimate activities that involve crypto wallets and currency.

* Non-production builds: For example, an extension that is still in an experimental stage, prone to crashes or instability and any extension that is designed for internal testing purposes only and not intended for public release.

* Prohibited products: An extension that promotes or facilitates the sale of illegal drugs, firearms, or counterfeit goods would be considered a prohibited product and also any extension that violates intellectual property rights, such as offering pirated software or copyrighted content without authorization, would be restricted.

* Gambling content.

<!-- ====================================================================== -->
## 1. Product policies

> <!-- todo: add content between headings -->

Author: 2.  Style guides say need content between headings. but these headings are a numbered outline, so does that apply? what could we put here between headings?

Author: it's an outline, keeping as-is

All functionality of extensions should be clearly disclosed to the user. Any extensions that attempt to deceive or mislead users will be removed from the store.

###### 1.1.3 Functionality

Your extension must be fully functional as in there should not be any blockers including, but not limited to, broken URLs, blocked links and pages.

###### 1.1.4 Search and discovery

Search terms may not exceed seven unique terms and should be relevant to your extension.

###### 1.1.5 Provide appropriate details

There should be distinct and informative details about your extension and the functionality in listing (metadata) for your extension.

###### 1.1.6 Stability and performance

Developers must ensure that their extensions are stable, do not cause the browser to freeze or crash.

In general, the extension cannot interfere with the normal functioning of the browser or operating system.

###### 1.1.7 Obfuscation

Extensions with obfuscated code aren't allowed.  This includes code within your extension package as well as any external code or resource fetched from the web.  You may be asked to refactor parts of your code if it is not reviewable.

###### 1.1.8 Altering browser settings

> 

Reviewer: "favorites"

> Your extension must not, without appropriate user consent, alter, or appear to alter, browser functionality or settings including, but not limited to: the address bar search provider and suggestions, the start or home page, the new tab page, and adding or removing favorite.

Reviewer: "favorites"

Author: the new tab page, and adding or removing favorites.

Extensions must declare all permissions required for their functionality in the manifest.  Your Extension must only request those permissions that are essential for functioning.

Declarative Net Request rules must be clearly declared within the extension and should not be imported remotely unless necessary for functionality.

Extensions must not collect or track sensitive user information without clear consent and must clearly state any user data handling in their privacy policy and user sensitive data must be encrypted with proper mechanisms like SSL and not transported via insecure channels.

Extensions must not promote deceptive installation of other extensions.

Bulk submissions of extensions with the same functionality and code are not allowed.

Extensions should not exploit security vulnerabilities of other extensions or applications and must keep authentication information secure.

Extensions that function as crypto wallets can only be published by verified company accounts

In general, your extension must not jeopardize or compromise user security, or the security or functionality of the device, system or related systems.

###### 1.2.1 Content security policies

> If you make any changes to your extension beyond the described functionality, any changes to code must be compliant with the [Microsoft Edge content security policy](csp.md#relaxing-the-default-policy).  For example, your extension should not download a remote script and subsequently run that script in a manner that is not consistent with the described functionality.

Author: Link text needs to match destination title/heading.  Linking to Relaxing section seems mistake.  Changed to:
```
If you make any changes to your extension beyond the described functionality, 
any changes to code must be compliant with 
the Content Security Policy for Microsoft Edge extensions; 
see [Content Security Policy (CSP)](./csp.md).  
For example, your extension should not download a remote script and subsequently 
run that script in a manner that is not consistent with the described functionality.
```

> If you make any changes to your extension beyond the described functionality, any changes to code must be compliant with the [Microsoft Edge content security policy](csp.md#relaxing-the-default-policy).  For example, your extension should not download a remote script and subsequently run that script in a manner that is not consistent with the described functionality.

Author: ```suggestion
If you make any changes to your extension beyond the described functionality, any changes to code must be compliant with the Content Security Policy for Microsoft Edge extensions; see [Content Security Policy (CSP)](./csp.md).  For example, your extension should not download a remote script and subsequently run that script in a manner that is not consistent with the described functionality.
```
* Link text needs to match destination title/heading.
* Linking to Relaxing section seems a mistake.  @Bhuvana-Priya

###### 1.2.2 Unwanted and malicious software

Your extension must not contain or enable malware that may harm the operation of the network, servers, and infrastructure of Microsoft or any third parties. Spyware, malicious scripts, and phishing scams are also prohibited.

> Your extension must comply with the Microsoft criteria for unwanted and malicious software, listed in [How Microsoft identifies malware and potentially unwanted applications](/windows/security/threat-protection/intelligence/criteria).

Reviewer: It must comply with the Microsoft criteria for Unwanted and Malicious Software.

Author: Capitalizing "Unwanted and Malicious Software" is incorrect, because the destination article https://review.learn.microsoft.com/en-us/microsoft-365/security/intelligence/criteria
doesn't mention the phrase.  It's not a formal proper-name phrase, so should not be capitalized.

It's not clear what you are proposing for linking.  well-formed Markdown is: `[link text](destination url)`

Author: keeping as-is.  Reviewers can enter a fresh comment and then click the Add Suggestion [+/-] button, and enter the change as Markdown tagging.

The _Microsoft Edge Add-ons store_ is also called the _Microsoft Edge Add-ons website_.

<!-- ====================================================================== -->
## Principles

A few principles to get you started:

* Offer unique and distinct value within your extensions for Microsoft Edge.  Provide a compelling reason to download your extensions from the Microsoft Edge Add-ons store (Microsoft Edge Add-ons).

* Do not mislead users about what your extension does, who is offering it, and so on.

* Do not attempt to cheat users, the system or the ecosystem.  There is no place in Microsoft Edge Add-ons for any kind of fraud; be it ratings and review manipulation, credit card fraud or other fraudulent activity.

Adhering to the Microsoft Edge Add-ons store developer policies helps you make choices that enhance the appeal and audience of your extension.

> 
> Your extensions are crucial to the experience of hundreds of millions of users.  We look forward to experiencing what you create and are thrilled to help deliver your extensions to the world.<!-- todo: delete paragraph?  not a principle, not essential, not substantive - fits better in Intro section.  focuses on "we in relation to you", not on technology or policies -->

Author: ```suggestion
```

Author: if the article has a statement like this, it would go in Intro section, more than Principles section

Author: removed the paragraph

ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 12/13/2023
---
# Microsoft Edge Add-ons store developer policies

> Thank you<!-- todo: go easy on "thank you", per corp-level style guide --> for your interest in developing extensions for the Microsoft Edge Add-ons store.  The Microsoft Edge Add-ons store developer policies (Add-ons store developer policies) apply to your extensions, including your submission of extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and the provision of such extensions through the Microsoft Edge Add-ons.

Author: repo hit counts:
please - 31 in 27 files
thank you - 10 in 10 files
thanks - 3 in 3 files

Author: used wording for the intro paragraph:
To develop an extension to be published through the Microsoft Edge Add-ons store, follow these Microsoft Edge Add-ons store developer policies (_Add-ons store developer policies_), which apply to submitting extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and providing extensions through the Microsoft Edge Add-ons store.

> Thank you<!-- todo: go easy on "thank you", per corp-level style guide --> for your interest in developing extensions for the Microsoft Edge Add-ons store.  The Microsoft Edge Add-ons store developer policies (Add-ons store developer policies) apply to your extensions, including your submission of extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and the provision of such extensions through the Microsoft Edge Add-ons.

Author: ```suggestion
Thanks for your interest in developing extensions for the Microsoft Edge Add-ons store.  The Microsoft Edge Add-ons store developer policies (Add-ons store developer policies) apply to your extensions, including your submission of extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and the provision of such extensions through the Microsoft Edge Add-ons.
```

Author: Writer/Editor proposal: strike "Thanks for your interest in developing extensions for the Microsoft Edge Add-ons store." - it's tangential and not substantive for this article; it's low-utility info content.  Then smooth out the remainder of intro paragraph.

Author: used wording:
To develop an extension to be published through the Microsoft Edge Add-ons store, follow these Microsoft Edge Add-ons store developer policies (_Add-ons store developer policies_), which apply to submitting extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and providing extensions through the Microsoft Edge Add-ons store.

> The _Microsoft Edge Add-ons store_ is also called the _Microsoft Edge Add-ons website_.  The _Microsoft Edge Add-ons store developer policies_ are also called the _Microsoft Edge Add-ons website developer policies_.

Author: ```suggestion
The _Microsoft Edge Add-ons store_ is also called the _Microsoft Edge Add-ons website_.  The _Microsoft Edge Add-ons store developer policies_ are also called the _Add-ons store developer policies_ or the _Microsoft Edge Add-ons website developer policies_.
```

Author: updated link text from website to store, then deleted:
The _Microsoft Edge Add-ons store developer policies_ are also called the _Add-ons store developer policies_ or the _Microsoft Edge Add-ons website developer policies_.

Your extension and associated metadata must clearly provide information about the types of in-product purchases offered and the range of prices. You must not mislead users and must be clear about the nature of your in-product promotions and offerings including the scope and terms of any trial experiences.

If your extension restricts access to user-created content during or after a trial, then you must notify users in every step of the process.

In addition, your extension must make it clear to users that they are initiating a purchase option in your extension.

<!-- ------------------------------ -->
#### 1.9 Notifications

Your extension must respect system settings for notifications. Any presentation of ads and notifications to users must be consistent with user preferences, regardless of whether the notifications are provided by the Microsoft Push Notification Service (MPNS), Windows Push Notification Service (WNS), or any other service.

If the user disables notifications, either on a product-specific or system-wide basis, then your extension must adhere to the user's notification settings and remain functional even though notifications may be disabled.

> You'll be receiving notifications related to your accounts, including but not limited to, login attempts, account changes, and security alerts.  To ensure the security of your accounts and the add-ons you submit, verify your accounts through a secure and approved method.

Reviewer: Also, you will be receiving notifications

Author: added Also,

###### 1.9.2 Ownership of notifications

You must not obscure or try to disguise the source of any notification initiated from your extension.

###### 1.9.3 No confidential or sensitive information

You must not include in a notification any information users may reasonably consider confidential or sensitive.

###### 1.9.4 Purpose of notifications

Notifications sent from your extension must relate to that extension or to other extensions you publish in the Microsoft Edge Add-ons store, and must not include promotional messages of any kind that aren't related to your extensions.

<!-- ------------------------------ -->
#### 1.10 Advertising conduct and content

For all advertising related activities, the following requirements apply:

###### 1.10.1 Purpose

The primary content of your extension must not be advertising, and your extension should not bundle other extension offers within the same flow.

Users should not be misled to click on advertisements to fully use the extension.

###### 1.10.2 Policies and agreements

Any advertising content your extension displays must adhere to [Microsoft Advertising Network policies](https://about.ads.microsoft.com/policies/home).

> If your extension displays ads, all content displayed must conform to the advertising requirements of the [App Developer Agreement](/legal/windows/agreements/app-developer-agreement) and this Policy.

Author: what is "this Policy" - 
* these Microsoft Edge Add-ons store developer policies; entire article
* policy section #  1.10.2 (Policies and agreements)

@Bhuvana-Priya

> If your extension displays ads, all content displayed must conform to the advertising requirements of the [App Developer Agreement](/legal/windows/agreements/app-developer-agreement) and this Policy.

Author: ```suggestion
If your extension displays ads, all content displayed must conform to the advertising requirements of the [App Developer Agreement](/legal/windows/agreements/app-developer-agreement) and the Microsoft Edge Add-ons store developer policies.
```

Author: converted the two paragraphs to a list

###### 1.10.3 Quality of advertising

* The primary purpose of your extension must not be to get users to click ads.

* Your extension must not do anything that interferes with or diminishes the visibility, value, or quality of any ads that it does display.

###### 1.10.4 Promotions

> If you purchase or create promotional ad campaigns to promote your extensions through the ad campaign functionality in [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd), all ad materials you provide to Microsoft, including any associated landing pages, must comply with [Microsoft Creative Specifications Policy](https://about.ads.microsoft.com/policies/creative-specs) and [Microsoft Advertising Network policies](https://about.ads.microsoft.com/policies/home).

Author: Partner Center is linked 4 of 5 mentions, yet there were 0 links to the Edge Add-ons site.  Linked the 2nd mention of the Edge Add-ons site, next to 1st mention of Partner Center, both linked:

To develop an extension to be published through the Microsoft Edge Add-ons store, follow these Microsoft Edge Add-ons store developer policies (_Add-ons store developer policies_), which apply to submitting extensions through [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and providing extensions through the [Microsoft Edge Add-ons](https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home) store.

> If you purchase or create promotional ad campaigns to promote your extensions through the ad campaign functionality in [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd), all ad materials you provide to Microsoft, including any associated landing pages, must comply with [Microsoft Creative Specifications Policy](https://about.ads.microsoft.com/policies/creative-specs) and [Microsoft Advertising Network policies](https://about.ads.microsoft.com/policies/home).

Author: unlinked

###### 1.10.5 Notifying users of opt-out for interest-based advertising

Your privacy statement or terms of use must let users know you plan to send Personal Information to the ad service provider and must tell users how they may opt-out of interest-based advertising.

###### 1.10.6 Other guidelines

> If your extension is directed at children under the age of 13, as defined in the [Children's Online Privacy Protection Act](https://www.ftc.gov/tips-advice/business-center/privacy-and-security/children%27s-privacy); you must notify Microsoft of this fact in [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and ensure that all ad content displayed in your extension is appropriate for children under the age of 13.

Author: ```suggestion
If your extension is directed at children under the age of 13, as defined in the [Children's Online Privacy Protection Act](https://www.ftc.gov/tips-advice/business-center/privacy-and-security/children%27s-privacy); you must notify Microsoft of this fact in [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd) and ensure that all ad content displayed in your extension is appropriate for children under the age of 13.
```
"Partner Center" occurs 5 times in this article.
4 times are linked - that's excessive linking; linking every mention for no reason.  Recommend un-linking some of the 4 instances.

And they are not a short Partner Center URL, they are a link that took a long time to resolve for me, and then rejected my account:
https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd

Author: unlinked

<!-- ====================================================================== -->
## 2 Content policies

> The following policies apply to content and metadata (including publisher name, extension name, extension icon, extension description, extension screenshots, extension trailers and trailer thumbnails, and any other extension metadata) offered for distribution in the Microsoft Edge Add-ons store.

Reviewer: remove "store"

Author: @Bhuvana-Priya - the sentence doesn't make sense without "store" at the end:
The following policies apply to content and metadata (including publisher name, extension name, extension icon, extension description, extension screenshots, extension trailers and trailer thumbnails, and any other extension metadata) offered for distribution in the Microsoft Edge Add-ons.

"for distribution in the Microsoft Edge Add-ons." wouldn't make sense, needs "store." at end

Author: If we link to the store, such that "Microsoft Edge Add-ons" becomes link text, omitting "store" or "website" might work better.

repo contains 23 hits in 12 files on the path:
https://microsoftedge.microsoft.com/addons/
![image](https://github.com/MicrosoftDocs/edge-developer/assets/45407486/290144f1-0d6c-4af1-a51a-208f147dbe96)

That short url redirects to:
https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home - 13 hits in 7 files:
![image](https://github.com/MicrosoftDocs/edge-developer/assets/45407486/28ff3994-ae3a-4849-a42c-bd793e1d2d86)

Author: Changed to:
The following policies apply to content and metadata (including publisher name, extension name, extension icon, extension description, extension screenshots, extension trailers and trailer thumbnails, and any other extension metadata) offered for distribution in [Microsoft Edge Add-ons](https://microsoftedge.microsoft.com/addons/Microsoft-Edge-Extensions-Home).

Content means the images, sounds, videos, and text contained in your extension, the tiles, notifications, error messages, or ads exposed through your extension, and anything delivered from a server or to which your extension connects.

Because extensions and Microsoft Edge Add-ons are used around the world, these requirements are interpreted and applied in the context of regional and cultural norms

<!-- ------------------------------ -->
#### 2.1 Content requirements

For your extension to be listed in the Microsoft Edge Add-ons store, your extension and metadata you submit should not contain mature, explicit, or otherwise inappropriate content.

This includes content that is:
* Discriminatory, hateful, or offensive.
* Harmful or dangerous.
* Misleading or fraudulent.
* Spam or malware.

> Examples of specific types of content that would be considered inappropriate for Microsoft Edge Add-ons include:

Reviewer: for Edge Add-Ons

Author: would be considered inappropriate for Edge Add-ons include:

#### 2.9 Illegal activity

Your extension and the associated data must not contain content or functionality that encourages, facilitates, or glamorizes illegal activity in the real world.

<!-- ------------------------------ -->
#### 2.10 Excessive profanity and inappropriate content

* Your extension must not contain excessive or gratuitous profanity.

* Your extension must not contain or display content that a reasonable person considers obscene.

<!-- ------------------------------ -->
#### 2.11 Country- or region-specific requirements

Content that is offensive in any country or region to which your extension is targeted is not allowed.  Content may be considered offensive in certain countries or regions because of local laws or cultural norms.  Examples of potentially offensive content in certain countries or regions include the following:

**China**

* Prohibited sexual content.
* Disputed territory or region references.
* Providing or enabling access to content or services that are illegal under applicable local law.

<!-- ------------------------------ -->
#### 2.12 Age ratings

<!-- todo: add content between headings -->

###### 2.12.1 Mature content

> When you submit your extension to [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd), you must indicate whether your extension displays content that should be marked "Mature".  When determining the rating for your extension, consider all the content in your app, including user generated content and ads, and to the content that your extension links.  If you indicate that your extension doesn't contain any "Mature" content, you are responsible for maintaining the accuracy of this rating.

Author: ```suggestion
When you submit your extension to Partner Center, you must indicate whether your extension displays content that should be marked "Mature".  When determining the rating for your extension, consider all the content in your app, including user generated content and ads, and to the content that your extension links.  If you indicate that your extension doesn't contain any "Mature" content, you are responsible for maintaining the accuracy of this rating.
```
instance 4

"Partner Center" occurs 5 times in this article.
4 times are linked - that's excessive linking; linking every mention for no reason.  Recommend un-linking some of the 4 instances.

And they are not a short Partner Center URL, they are a link that took a long time to resolve for me, and then rejected my account:
https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd

Reviewer 2: @Bhuvana-Priya can we shorten the URL

Author: My results from URL lengths:

https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd - after a long time, i end up at https://partner.microsoft.com/en-us/dashboard/registration/AccountInfo?accountProgram=MSEdgeAddons "This account is not registered with Microsoft Edge program on Partner Center"
https://partner.microsoft.com/dashboard/microsoftedge/public/login - after a long time, i end up at https://partner.microsoft.com/en-us/dashboard/registration/AccountInfo?accountProgram=MSEdgeAddons "This account is not registered with Microsoft Edge program on Partner Center"

https://partner.microsoft.com/dashboard/microsoftedge/public - after a long time, i end up at https://partner.microsoft.com/en-us/dashboard/microsoftedge/

https://partner.microsoft.com/dashboard/microsoftedge - after a long time, i end up at https://partner.microsoft.com/en-us/dashboard/home
https://partner.microsoft.com/dashboard - after a long time, i end up at https://partner.microsoft.com/en-us/dashboard/home

https://partner.microsoft.com - quickly, end up at a regular webpage: https://partner.microsoft.com/en-US/  "Limitless innovation with Microsoft AI - Welcome to the Microsoft AI Cloud Partner Program."

Author: @Bhuvana-Priya @AbhisekhMSFT - 
In the Intro paragraph, I changed the Partner Center url from:
https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd
to:
https://partner.microsoft.com/dashboard/home
because that tagged URL matches where i end up, and that page contains some content for me instead of being empty.  It seems to be the best general-purpose link into Partner Center.

I un-linked the other mentions of Partner Center in this article.

Regardless of the rating given to your extension, it must still adhere to all the content requirements of the Microsoft Edge Add-ons store developer policies.

###### 2.12.2 Ratings change

If your extension provides content (such as user-generated, retail, or other web-based content) that might be appropriate for a higher age rating than the assigned rating, you must require users to opt into receiving such content by using a content filter or by signing in with a pre-existing account.

<!-- ------------------------------ -->
#### 2.13 Videos

> If you submit a promotional video in the listing, it should follow all the content guidelines mentioned in this policy.  If you choose to provide a YouTube link, you must ensure that advertisements are turned off for the specific videos that you want to embed.  For more information on turning on or off ads on YouTube, see [YouTube advertising formats](https://support.google.com/youtube/answer/2531367?ref_topic=7072227) and [Ads on embedded videos](https://support.google.com/youtube/answer/132596).

Reviewer: "If you submit a promotional video in the listing, then it should strictly adhere to our content guidelines." ``

Author: Revising to that, although the revision reads more awkwardly than the existing text:
* "our" content guidelines has objectionable tone, overly personal, and is less specific than "the content guidelines mentioned in this policy".
* "then" is superfluous and doesn't read as well/as tightly.

<!-- ====================================================================== -->
## Complaint and appeal process for Microsoft Edge Add-ons certification 

> All extensions must adhere to the store policies listed above.  If your extension failed in the review process, review the store policies to understand the reason for failure.  After submitting your extension using Partner Center, to ask a question about the review or certification status of it, go to [New Support Request](https://support.microsoft.com/supportrequestform/e7a381be-9c9a-fafb-ed76-262bc93fd9e4) and complete the form.

Author: "Partner Center" occurs 5 times in this article.
4 times are linked - that's excessive linking; linking every mention for no reason.  The present instance is not linked, and that works well.  Recommend un-linking some of the 4 instances.  

And they are not a short Partner Center URL, they are a link that took a long time to resolve for me, and then rejected my account:
https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd

Author: keeping this instance non-linked

> All extensions must adhere to the store policies listed above.  If your extension failed in the review process, review the store policies to understand the reason for failure.  After submitting your extension using Partner Center, to ask a question about the review or certification status of it, go to [New Support Request](https://support.microsoft.com/supportrequestform/e7a381be-9c9a-fafb-ed76-262bc93fd9e4) and complete the form.

Author: the url 
`[New Support Request](https://support.microsoft.com/supportrequestform/e7a381be-9c9a-fafb-ed76-262bc93fd9e4)`
https://support.microsoft.com/supportrequestform/e7a381be-9c9a-fafb-ed76-262bc93fd9e4
redirects to
https://support.serviceshub.microsoft.com/supportforbusiness/onboarding?origin=/supportforbusiness/create
on my machine.

* Pornography or nudity.
* Hate speech or symbols.
* Violence or gore.
* Drugs or alcohol.
* Gambling.
* Illegal activities.
* Misleading or fraudulent claims.
* Spam or malware.

The above list of inappropriate content is not exhaustive.  The Microsoft Edge Add-ons store may reject or remove any extension that they deem to be inappropriate, even if it does not violate any specific policy.

An extension that doesn't meet the requirements for Microsoft Edge Add-ons store listings may be rejected or promptly removed.

<!-- ------------------------------ -->
#### 2.2 Content including names, logos, and original and third-party content

All content in your extension and associated metadata must be originally created by you or appropriately licensed from a rights holder and must be used only as permitted by the rights holder or as otherwise permitted by law.

The content must reference Microsoft trademarks only to the extent necessary and should not modify or alter the trademarks in any way.

If not created or appropriately licensed, a disclaimer should be added stating the same

<!-- ------------------------------ -->
#### 2.3 Risk of harm

> <!-- todo: add content between headings if corp-level style guide applies here -->

Author: 3. style guides say need content between headings. but these headings are a numbered outline, so does that apply? what could we put here between headings?

Author: it's an outline, keeping as-is

<!-- ------------------------------ -->
#### 2.9 Illegal activity

Your extension and the associated data must not contain content or functionality that encourages, facilitates, or glamorizes illegal activity in the real world.

<!-- ------------------------------ -->
#### 2.10 Excessive profanity and inappropriate content

* Your extension must not contain excessive or gratuitous profanity.

* Your extension must not contain or display content that a reasonable person considers obscene.

<!-- ------------------------------ -->
#### 2.11 Country- or region-specific requirements

Content that is offensive in any country or region to which your extension is targeted is not allowed.  Content may be considered offensive in certain countries or regions because of local laws or cultural norms.  Examples of potentially offensive content in certain countries or regions include the following:

**China**

* Prohibited sexual content.
* Disputed territory or region references.
* Providing or enabling access to content or services that are illegal under applicable local law.

<!-- ------------------------------ -->
#### 2.12 Age ratings

> <!-- todo: add content between headings if corp-level style guide applies here -->

Author: style guides say need content between headings.  but these headings are a numbered outline, so does that apply?  what could we put here between headings?

Author: see various examples near https://learn.microsoft.com/en-us/copilot-plugins/publish/store-policies#4-product-and-quality-guidelines

Author: it's an outline format; keeping as-is

###### 2.12.1 Mature content

When you submit your extension to [Partner Center](https://partner.microsoft.com/dashboard/microsoftedge/public/login?ref=dd), you must indicate whether your extension displays content that should be marked "Mature".  When determining the rating for your extension, consider all the content in your app, including user generated content and ads, and to the content that your extension links.  If you indicate that your extension doesn't contain any "Mature" content, you are responsible for maintaining the accuracy of this rating.

Regardless of the rating given to your extension, it must still adhere to all the content requirements of the Microsoft Edge Add-ons store developer policies.

###### 2.12.2 Ratings change

If your extension provides content (such as user-generated, retail, or other web-based content) that might be appropriate for a higher age rating than the assigned rating, you must require users to opt into receiving such content by using a content filter or by signing in with a pre-existing account.

<!-- ------------------------------ -->
#### 2.13 Videos

> If you submit a promotional video in the listing, it should follow all the content guidelines mentioned in this policy.  If you choose to provide a YouTube link, you must ensure that advertisements are turned off for the specific videos that you want to embed.  For more information on turning on or off ads on YouTube, see [Set your default ad formats](https://support.google.com/youtube/answer/2531367?ref_topic=7072227) and [Ads on embedded videos](https://support.google.com/youtube/answer/132596).

Author: link text "Set your default ad formats" is wrong or outdated;
https://support.google.com/youtube/answer/2531367?ref_topic=7072227
```suggestion
If you submit a promotional video in the listing, it should follow all the content guidelines mentioned in this policy.  If you choose to provide a YouTube link, you must ensure that advertisements are turned off for the specific videos that you want to embed.  For more information on turning on or off ads on YouTube, see [YouTube advertising formats](https://support.google.com/youtube/answer/2531367?ref_topic=7072227) and [Ads on embedded videos](https://support.google.com/youtube/answer/132596).
```
has title: "YouTube advertising formats"

<!-- ------------------------------ -->
#### Appeal statistics for Microsoft Edge Add-ons for FY2022

| Statistic                                                             | Count     |
|-----------------------------------------------------------------------|-----------|
| App and/or account enforcement action appeals                         |        34 |
| Complaints about technological issues                                 |         0 |
| Regulatory compliance complaints                                      |         0 |
| Questions about certification, policy, submission, and technical help |       131 |
| Miscellaneous                                                         |         5 |
| Total issues                                                          |       175 |
| Overturned decisions                                                  |        28 |
| Average processing time                                               | 6.58 days |

> These numbers were reported on July 1, 2023|January 7, 2023.<!-- todo: which? -->

Author: @Bhuvana-Priya - were these numbers reported on July 1, or January 7?

Reviewer 2: This appeal stats : July 1

Author: These numbers were reported on July 1, 2023.
